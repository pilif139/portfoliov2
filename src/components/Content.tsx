import Heading from "@/components/ui/Heading"
import { PostContentBlock } from "@/db/schema/posts"
import Image from "next/image"

export default function Content({ content }: { content: Partial<PostContentBlock> }) {
    switch (content.type) {
        case "h1":
            return <Heading variant="1">{content.content}</Heading>
        case "h2":
            return <Heading variant="2">{content.content}</Heading>
        case "h3":
            return <Heading variant="3">{content.content}</Heading>
        case "h4":
            return <Heading variant="4">{content.content}</Heading>
        case "p":
            return <p>{content.content}</p>
        case "image":
            return (
                <Image
                    src={content.content as string}
                    alt="img"
                    width={400}
                    height={400}
                    className="shadow-gray-900 shadow-lg max-w-min max-h-min"
                />
            )
        case "video":
            return (
                <video src={content.content as string} controls width="350" height="350" className="shadow-gray-900 shadow-lg" />
            )
        case "file":
            return (
                <a href={content.content as string} download className="bg-theme-10 px-4 py-2 rounded-xl">
                    Download file
                </a>
            )
        default:
            return <></>
    }
}
