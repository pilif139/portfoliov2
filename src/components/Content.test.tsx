import { render, screen } from "@testing-library/react"
import Content from "./Content"

jest.mock("next/image", () => ({ src, alt }: { src: string; alt: string }) => <img src={src} alt={alt} />)

describe("Content component", () => {
    it("should render h1", () => {
        const { container } = render(<Content content={{ type: "h1", content: "Hello" }} />)
        expect(container).toMatchSnapshot()
    })

    it("should render h2", () => {
        const { container } = render(<Content content={{ type: "h2", content: "Hello" }} />)
        expect(container).toMatchSnapshot()
    })

    it("should render h3", () => {
        const { container } = render(<Content content={{ type: "h3", content: "Hello" }} />)
        expect(container).toMatchSnapshot()
    })

    it("should render h4", () => {
        const { container } = render(<Content content={{ type: "h4", content: "Hello" }} />)
        expect(container).toMatchSnapshot()
    })

    it("should render p", () => {
        const { container } = render(<Content content={{ type: "p", content: "Hello" }} />)
        expect(container).toMatchSnapshot()
    })

    it("should render image", () => {
        const { container } = render(<Content content={{ type: "image", content: "image.jpg", altText: "alternative text" }} />)
        expect(container).toMatchSnapshot()
        expect(screen.getByAltText("alternative text")).toBeInTheDocument()
    })

    it("should render video", () => {
        const { container } = render(<Content content={{ type: "video", content: "video.mp4" }} />)
        expect(container).toMatchSnapshot()
    })

    it("should render anchor tag", () => {
        const { container } = render(<Content content={{ type: "file", content: "file.pdf" }} />)
        expect(container).toMatchSnapshot()
        expect(screen.getByText("Download file")).toBeInTheDocument()
    })

    it("should render nothing", () => {
        const { container } = render(<Content content={{ content: "Hello" }} />)
        expect(container.firstChild).toBeNull()
    })
})
