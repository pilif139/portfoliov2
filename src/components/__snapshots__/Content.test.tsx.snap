// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content component should render anchor tag 1`] = `
<div>
  <a
    class="bg-theme-10 px-4 py-2 rounded-xl"
    download=""
    href="file.pdf"
  >
    Download file
  </a>
</div>
`;

exports[`Content component should render h1 1`] = `
<div>
  <h1
    class="text-5xl font-bold text-theme-10 undefined"
  >
    Hello
  </h1>
</div>
`;

exports[`Content component should render h2 1`] = `
<div>
  <h2
    class="text-4xl font-bold text-theme-10 undefined"
  >
    Hello
  </h2>
</div>
`;

exports[`Content component should render h3 1`] = `
<div>
  <h3
    class="text-3xl font-bold text-theme-10 undefined"
  >
    Hello
  </h3>
</div>
`;

exports[`Content component should render h4 1`] = `
<div>
  <h4
    class="text-2xl font-bold text-theme-10 undefined"
  >
    Hello
  </h4>
</div>
`;

exports[`Content component should render image 1`] = `
<div>
  <img
    alt="alternative text"
    src="image.jpg"
  />
</div>
`;

exports[`Content component should render p 1`] = `
<div>
  <p>
    Hello
  </p>
</div>
`;

exports[`Content component should render video 1`] = `
<div>
  <video
    class="shadow-gray-900 shadow-lg"
    controls=""
    height="350"
    src="video.mp4"
    width="350"
  />
</div>
`;
